common/datetime.c

```diff
@@ -265,3 +257,78 @@ void oe_datetime_log(const char* msg, const oe_datetime_t* date)
         OE_TRACE_VERBOSE("%s %s\n", msg, str);
     }
 }
+
+oe_result_t oe_datetime_to_time_t(const oe_datetime_t* datetime, time_t* value)
+{
+    oe_result_t result = OE_FAILURE;
+    time_t tmp = 0;
+    struct tm timeinfo = {0};
+
+    if (datetime == NULL || value == NULL)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    OE_CHECK(oe_datetime_is_valid(datetime));
+
+    // Update timeinfo based on input oe_date_time
+    timeinfo.tm_year = (int)datetime->year - 1900;
+    timeinfo.tm_mon = (int)datetime->month - 1;
+    timeinfo.tm_mday = (int)datetime->day;
+    timeinfo.tm_hour = (int)datetime->hours;
+    timeinfo.tm_min = (int)datetime->minutes;
+    timeinfo.tm_sec = (int)datetime->seconds;
+
+    tmp = mktime(&timeinfo);
+
+    *value = tmp;
+
+    result = OE_OK;
+done:
+    return result;
+}
+
+oe_result_t oe_time_t_to_datetime(const time_t* value, oe_datetime_t* datetime)
+{
+    oe_result_t result = OE_FAILURE;
+    struct tm timeinfo = {0};
+
+    if (value == NULL || datetime == NULL)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    gmtime_r(value, &timeinfo);
+
+    datetime->year = (uint32_t)timeinfo.tm_year + 1900;
+    datetime->month = (uint32_t)timeinfo.tm_mon + 1;
+    datetime->day = (uint32_t)timeinfo.tm_mday;
+    datetime->hours = (uint32_t)timeinfo.tm_hour;
+    datetime->minutes = (uint32_t)timeinfo.tm_min;
+    datetime->seconds = (uint32_t)timeinfo.tm_sec;
+
+    result = OE_OK;
+done:
+    return result;
+}
+
+oe_result_t oe_gmt_to_localtime(const time_t* gmtime, time_t* localtime)
+{
+    oe_result_t result = OE_FAILURE;
+    time_t now = 0;
+    time_t utc = 0;
+    time_t time_diff = 0;
+    struct tm timeinfo = {0};
+
+    if (localtime == NULL || gmtime == NULL)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    now = time(NULL);
+    gmtime_r(&now, &timeinfo);
+
+    utc = mktime(&timeinfo);
```

Comment:
This is the code that I don't understand how it's secure.   I believe mktime internally calls do_tzset() which gets the time zone information from the "TZ" environment variable or the "/etc/localtime" file.
And time() in line 322 results in an ocall and so the returned value is similarly untrusted.

Reply:
All the operation should be done inside enclave, so I think it's secure.